packages/docmanager-extension/src/index.ts

```diff
@@ -735,6 +752,42 @@ function addCommands(
     }
   });
 
+  commands.addCommand(CommandIDs.toggleNameFileOnSave, {
+    label: trans.__('Name File on First Save'),
+    isToggled: () => docManager.nameFileOnSave,
+    execute: () => {
+      const value = !docManager.nameFileOnSave;
+      const key = 'nameFileOnSave';
+      return settingRegistry
+        .set(pluginId, key, value)
+        .catch((reason: Error) => {
+          console.error(`Failed to set ${pluginId}:${key} - ${reason.message}`);
+        });
+    }
+  });
+
+  docManager.optionChanged.connect(() => {
+    const key = 'nameFileOnSave';
+    const value = settingRegistry.plugins[pluginId]?.data.user[key];
+    if (value == docManager.nameFileOnSave) {
+      void settingRegistry.set(pluginId, key, !value).catch((reason: Error) => {
+        console.error(`Failed to set ${pluginId}:${key} - ${reason.message}`);
+      });
+    }
+  });
+
+  docManager.activateRequested.connect((sender, args) => {
+    const widget = sender.findWidget(args);
+    if (widget && widget instanceof NotebookPanel) {
```

Comment:
@jess-x shouldn't we be doing this for all files, not just notebooks? 

I am also wary of adding this dependency to docmanager: